onnxruntime/test/contrib_ops/shape_inference_test_helper.h

```diff
@@ -3,9 +3,47 @@
 
 #include "gtest/gtest.h"
 #include "test/providers/provider_test_utils.h"
+
+#ifdef __GNUC__
+#include "onnxruntime_config.h"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wignored-qualifiers"
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#ifdef HAS_DEPRECATED_DECLARATIONS
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+#else
+#pragma warning(push)
+#pragma warning(disable : 4018) /*'expression' : signed/unsigned mismatch */
+#pragma warning(disable : 4065) /*switch statement contains 'default' but no 'case' labels*/
+#pragma warning(disable : 4100)
+#pragma warning(disable : 4146) /*unary minus operator applied to unsigned type, result still unsigned*/
+#pragma warning(disable : 4127)
+#pragma warning(disable : 4244)  /*'conversion' conversion from 'type1' to 'type2', possible loss of data*/
+#pragma warning(disable : 4251)  /*'identifier' : class 'type' needs to have dll-interface to be used by clients of class 'type2'*/
+#pragma warning(disable : 4267)  /*'var' : conversion from 'size_t' to 'type', possible loss of data*/
+#pragma warning(disable : 4305)  /*'identifier' : truncation from 'type1' to 'type2'*/
+#pragma warning(disable : 4307)  /*'operator' : integral constant overflow*/
+#pragma warning(disable : 4309)  /*'conversion' : truncation of constant value*/
+#pragma warning(disable : 4334)  /*'operator' : result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)*/
+#pragma warning(disable : 4355)  /*'this' : used in base member initializer list*/
+#pragma warning(disable : 4506)  /*no definition for inline function 'function'*/
+#pragma warning(disable : 4800)  /*'type' : forcing value to bool 'true' or 'false' (performance warning)*/
+#pragma warning(disable : 4996)  /*The compiler encountered a deprecated declaration.*/
+#pragma warning(disable : 6011)  /*Dereferencing NULL pointer*/
+#pragma warning(disable : 6387)  /*'value' could be '0'*/
+#pragma warning(disable : 26495) /*Variable is uninitialized.*/
+#endif
+
```

Comment:
Would it make sense to put these in a file that gets included so we can put them in one place? e.g. disable_warnings_from_onnx.inc (could also have reenable_warnings.inc with the 'pop' pragmas to pair those .inc files together as that may be a little easier to understand what's going on)